_inc/lib/core-api/wpcom-fields/attachment-fields-videopress.php

```diff
@@ -0,0 +1,171 @@
+<?php
+/**
+ * Extend the REST API functionality for VideoPress users.
+ *
+ * @package Jetpack
+ */
+
+/**
+ * Add per-attachment VideoPress data.
+ *
+ * { # Attachment Object
+ *   ...
+ *   jetpack_videopress_guid: (string) VideoPress identifier
+ *   ...
+ * }
+ *
+ * @since 7.1.0
+ */
+class WPCOM_REST_API_V2_Attachment_VideoPress_Field extends WPCOM_REST_API_V2_Field_Controller {
+	/**
+	 * The REST Object Type to which the jetpack_videopress_guid field will be added.
+	 *
+	 * @var string
+	 */
+	protected $object_type = 'attachment';
+
+	/**
+	 * The name of the REST API field to add.
+	 *
+	 * @var string $field_name
+	 */
+	protected $field_name = 'jetpack_videopress_guid';
+
+	/**
+	 * Registers the jetpack_videopress field and adds a filter to remove it for attachments that are not videos.
+	 */
+	public function register_fields() {
+		parent::register_fields();
+
+		add_filter( 'rest_prepare_attachment', array( $this, 'remove_field_for_non_videos' ), 10, 2 );
+	}
+
+	/**
+	 * Defines data structure and what elements are visible in which contexts
+	 */
+	public function get_schema() {
+		return array(
+			'$schema'     => 'http://json-schema.org/draft-04/schema#',
+			'title'       => $this->field_name,
+			'type'        => 'string',
+			'context'     => array( 'view', 'edit' ),
+			'readonly'    => true,
+			'description' => __( 'Unique VideoPress ID', 'jetpack' ),
+		);
+	}
+
+	/**
+	 * Getter: Retrieve current VideoPress data for a given attachment.
+	 *
+	 * @param mixed           $object  Response from the attachment endpoint.
+	 * @param WP_REST_Request $request Request to the attachment endpoint.
+	 *
+	 * @return string
+	 */
+	public function get( $object, $request ) {
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			$blog_id = get_current_blog_id();
+		} else {
+			$blog_id = Jetpack_Options::get_option( 'id' );
+		}
+
+		$post_id = absint( $object['id'] );
+
+		$videopress_guid = $this->get_videopress_guid( $post_id, $blog_id );
+
+		if ( ! $videopress_guid ) {
+			return '';
+		}
+
+		return $videopress_guid;
+	}
+
+	/**
+	 * Gets the VideoPress GUID for a given attachment.
+	 *
+	 * This is pulled out into a separate method to support unit test mocking.
```

Comment:
you are so amazing - thanks for clearly indicating why this thin wrapper exists!